docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={(matchDetails) => <div>{`Home ${matchDetails.params.page}`}</div>} />
+			</div>
+		);
+	}
+}
+```
+
+Query parameters are also defined using curly braces split from the route path by a `?`. To define more than one query param, use the `&` delimiter.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}?{queryOne}&{queryOne}',
+		outlet: 'home'
+	}
+];
+```
+
+The query params are injected into matching `Outlet` `renderer` properties accessed from `queryParams` property.
+
+## Default Route & Parameters
+
+To add a default route for your application update the routing configuration to include `defaultRoute: true` for the preferred route.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	}
+];
+```
+
+If the default route has path or query parameters a map of defaults need to be specified.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}',
+		outlet: 'home',
+		defaultRoute: true,
+		defaultParams: {
+			page: 'about'
+		}
+	}
+];
+```
+
+## Route Configuration
+
+The routing configuration is a hierarchical structure used to describe the entire Dojo application, associating `outlet` ids to a routing path. The routing path can be nested using children which enables building a routing structure that can accurately reflect the requirements of the application.
+
+The routing configuration API is constructed with the following properties:
+
+* `path: string`: The routing path segment to match in the URL.
+* `outlet: string`: The `outlet` id used to render widgets to the associated routing path.
+* `defaultRoute: boolean`(optional): Marks the outlet as default, the application will redirect to this route automatically if an unknown routes is detected on application load.
+* `defaultParams: { [index: string]: string }`(optional): Associated default parameters (`path` and `query`), required if the default route has required params.
+* `children: RouteConfig[]`(optional): Nested child routing configuration.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	},
+	{
+		path: 'about',
+		outlet: 'about-overview',
+		children: [
+			{
+				path: '{services}',
+				outlet: 'about-services'
+			},
+			{
+				path: 'company',
+				outlet: 'about-company'
+			},
+			{
+				path: 'history',
+				outlet: 'about-history'
+			}
+		]
+	}
+];
+```
+
+This example would register the following routes and outlets:
+
+| Route             | Outlet          |
+| ----------------- | --------------- |
+| `/home`           | `home`          |
+| `/about`          | `about-overview`|
+| `/about/company`  | `about-company` |
+| `/about/history`  | `about-history` |
+| `/about/{param}`  | `about-services`|
```

Comment:
the param bit confuses me here, why is it not `services`?

Reply:
It was to demonstrate that any other value would match the `about-services` outlet, but I can clean that up and just add a couple of extra paths that both route to `about-services`

Reply:
yeah this table doesn't really make sense to me.

Reply:
It is supposed to show how a routing paths map to outlets.

Reply:
yeah like i say, i don't know what the left column is, is it the url? it just says route, but the route had `{services}`. real examples always help (ie the input).

Reply:
gotcha, I'll make some changes and see if it makes more sense.

Reply:
I changed the title to say `Url Path` and used multiple examples that both map to the `about-services` outlet. The changes can be seen in the `route-configuration.md` file.